### PR TITLE
Rename Parameter class in Topology to TopologyParameter

### PR DESCRIPTION
**Pull request by heplesser**
The renaming is motivated by #352 and the need for a `Parameter` class in NEST. As an extra, it fixes a little detail in the scripts used to build the topology user manual. The main point of renaming in topology is that the `Parameter` class to be introduced in nestkernel will be the general parameter class for the future.

@jakobj @jougs Would you take a look---it is really just renaming.

### CODE REVIEW

File: topology/doc/user_manual_scripts/connections.py [hunk 1]
```diff
@@ -24,6 +24,7 @@
 import nest
 import nest.topology as tp
 import matplotlib.pyplot as plt
+from mpl_toolkits.mplot3d.axes3d import Axes3D
```
Comment (jougs) [inline topology/doc/user_manual_scripts/connections.py]: Is this used somewhere in the file? If it is, why was this `import` not here before?

**Comment by jougs** [comment]
Why did you re-add all the figures? For me they look the same, with only very minor shifts probably due to another version of some font library. Can you please remove them?

The code changes are trivial and look good to me. :+1: once the images are gone and my other comment is addressed.

File: topology/doc/user_manual_scripts/connections.py (hunk 1 above)
Comment (heplesser) [inline topology/doc/user_manual_scripts/connections.py]: This is required to enable [3d projection here](https://github.com/heplesser/nest-simulator/blob/63fc287e0ba00d1c02c13bca4a4c8cbeea1c1352/topology/doc/user_manual_scripts/connections.py#L223). I am not sure why it was not needed before. That may be due to internal changes in matplotlib.

Comment (jougs) [inline topology/doc/user_manual_scripts/connections.py]: Thanks for the explanation.

**Comment by heplesser** [comment]
@jougs I committed all the figure files because they were recreated with the modified script and topology code. Incidentally, this fixes #300: Fig 3.6. is now back in the PDF. But I agree that the figures should probably not be committed, just the final manuscript PDF. I will create a new issue for that change. I'd appreciate merging this PR soon, since #352 depends on it.

**Comment by jougs** [comment]
Agreed. As this is just a simple rename otherwise, I'll merge without second review.

**Pull request merged**